pt_os_web_portal/backend/routes.py

```diff
@@ -594,9 +595,11 @@ def post_stop_vnc_wpa_gui():
 @app.route("/vnc-wpa-gui-url", methods=["GET"])
 def get_vnc_wpa_gui_url():
     logger.debug("Route '/vnc-wpa-gui-url'")
+    url = ""
     try:
         details = vnc_connection_details(PtWebVncDisplayId.WpaGui.value)
-        url = f"{request.remote_addr}:{details.port}/{details.path}"
+        if is_url(details.url):
+            url = f"{details.scheme}://{request.host}:{details.port}{details.path}"
```

Comment:
docs suggest that request.host might include a port sometimes:
https://flask.palletsprojects.com/en/1.1.x/api/#flask.Request.host